clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -296,9 +281,11 @@ public boolean awaitPendingRequests(Node node, long timeoutMs) {
      */
     public int pendingRequestCount(Node node) {
         synchronized (this) {
-            List<ClientRequest> pending = unsent.get(node);
-            int unsentCount = pending == null ? 0 : pending.size();
-            return unsentCount + client.inFlightRequestCount(node.idString());
+            int total = 0;
+            for (UnsentRequest request : unsent)
```

Comment:
@lindong28 Let's take a step back for a second. My concern with the initial patch was the addition of a second lock, which _does_ complicate reasoning about this class. There are basically two ways to address this:

1. My suggestion: remove the need for a second lock. You apparently were not satisfied with this solution and cited several shortcomings, but adopted it anyway. Now you are resisting my attempts to address its shortcomings.

2. Ismael's suggestion: provide better encapsulation of the second lock. Did you even try this?